### PR TITLE
Enabling BPFLSM based KSP protection on Kubearmor itself

### LINKED ISSUE #1688: mount procfs instead of using hostpid
## Feature Request

KubeArmor needs access to host's procfs to stich container and real time process information we get from eBPF. 

**Is your feature request related to a problem? Please describe the use case.**

Sharing pid ns with host is more risky than just accessing procfs.

**Describe the solution you'd like**

- We can mount and set procfs to readonly maybe alleviating a lot of risks involded with mounting procfs
- Add configuration in KubeArmor to support custom path prefix for procfs

Related #1186

### PR DESCRIPTION
**Pull request by daemon1024**
**Purpose of PR?**:

Succeeds #1717
Fixes #1688 

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

**Review by rksharma95 — DISMISSED** [review]
LGTM :+1:

File: KubeArmor/core/containerdHandler.go [hunk 1]
```diff
@@ -193,13 +193,13 @@ func (ch *ContainerdHandler) GetContainerInfo(ctx context.Context, containerID s
 
 		pid := strconv.Itoa(int(taskRes.Processes[0].Pid))
 
-		if data, err := os.Readlink("/proc/" + pid + "/ns/pid"); err == nil {
+		if data, err := os.Readlink(cfg.GlobalCfg.ProcFsMount + "/" + pid + "/ns/pid"); err == nil {
```
Comment (Aryan-sharma11) [inline KubeArmor/core/containerdHandler.go]:
```suggestion
		if data, err :=os.Readlink(filepath.Join(cfg.GlobalCfg.ProcFsMount, pid, "ns/pid")); err == nil {
```

File: KubeArmor/core/containerdHandler.go [hunk 2]
```diff
@@ -193,13 +193,13 @@ func (ch *ContainerdHandler) GetContainerInfo(ctx context.Context, containerID s
...
 			if _, err := fmt.Sscanf(data, "pid:[%d]\n", &container.PidNS); err != nil {
 				kg.Warnf("Unable to get PidNS (%s, %s, %s)", containerID, pid, err.Error())
 			}
 		}
 
-		if data, err := os.Readlink("/proc/" + pid + "/ns/mnt"); err == nil {
+		if data, err := os.Readlink(cfg.GlobalCfg.ProcFsMount + "/" + pid + "/ns/mnt"); err == nil {
```
Comment (Aryan-sharma11) [inline KubeArmor/core/containerdHandler.go]:
```suggestion
		if data, err := os.Readlink(filepath.Join(cfg.GlobalCfg.ProcFsMount, pid, "ns/pid")); err == nil {
```

File: KubeArmor/core/crioHandler.go [hunk 3]
```diff
@@ -130,15 +130,15 @@ func (ch *CrioHandler) GetContainerInfo(ctx context.Context, containerID string,
 
 	pid := strconv.Itoa(containerInfo.Pid)
 
-	if data, err := os.Readlink("/proc/" + pid + "/ns/pid"); err == nil {
+	if data, err := os.Readlink(cfg.GlobalCfg.ProcFsMount + "/" + pid + "/ns/pid"); err == nil {
```
Comment (Aryan-sharma11) [inline KubeArmor/core/crioHandler.go]:
```suggestion
	if data, err := os.Readlink(filepath.Join(cfg.GlobalCfg.ProcFsMount, pid, "ns/pid")); err == nil{
```

File: KubeArmor/core/crioHandler.go [hunk 4]
```diff
@@ -130,15 +130,15 @@ func (ch *CrioHandler) GetContainerInfo(ctx context.Context, containerID string,
...
 		if _, err := fmt.Sscanf(data, "pid:[%d]\n", &container.PidNS); err != nil {
 			kg.Warnf("Unable to get PidNS (%s, %s, %s)", containerID, pid, err.Error())
 		}
 	} else {
 		return container, err
 	}
 
-	if data, err := os.Readlink("/proc/" + pid + "/ns/mnt"); err == nil {
+	if data, err := os.Readlink(cfg.GlobalCfg.ProcFsMount + "/" + pid + "/ns/mnt"); err == nil {
```
Comment (Aryan-sharma11) [inline KubeArmor/core/crioHandler.go]:
```suggestion
	if data, err := os.Readlink( filepath.Join(cfg.GlobalCfg.ProcFsMount , pid , "/ns/mnt")); err == nil {
```

File: KubeArmor/core/dockerHandler.go [hunk 5]
```diff
@@ -144,13 +144,13 @@ func (dh *DockerHandler) GetContainerInfo(containerID string, OwnerInfo map[stri
 
 	pid := strconv.Itoa(inspect.State.Pid)
 
-	if data, err := os.Readlink("/proc/" + pid + "/ns/pid"); err == nil {
+	if data, err := os.Readlink(cfg.GlobalCfg.ProcFsMount + "/" + pid + "/ns/pid"); err == nil {
```
Comment (Aryan-sharma11) [inline KubeArmor/core/dockerHandler.go]:
```suggestion
	if data, err := os.Readlink(filepath.Join(cfg.GlobalCfg.ProcFsMount, pid ,"/ns/pid")); err == nil {
```

**Review by Aryan-sharma11 — CHANGES_REQUESTED** [review]
lets use `filepath.Join()` to build paths.

**Review by Aryan-sharma11 — APPROVED** [review]

**Review by rootxrishabh — APPROVED** [review]
LGTM :+1:

**Pull request merged**